src/terminal/buffer.c

```diff
@@ -164,7 +164,7 @@ void guac_terminal_buffer_copy_rows(guac_terminal_buffer* buffer,
         guac_terminal_buffer_row* dst_row = guac_terminal_buffer_get_row(buffer, current_row + offset, src_row->length);
 
         /* Copy data */
-        memcpy(dst_row->characters, src_row->characters, sizeof(guac_terminal_char) * src_row->length);
+        memcpy(dst_row->characters, src_row->characters, sizeof(guac_terminal_char) * src_row->length + 1);
```

Comment:
It doesn't look like the storage space in `dst_row->characters` actually got any bigger - you'll need to allocate additional space for this newly added character.

Reply:
Or rather, even better, this should probably just be a completely independent flag on the `guac_terminal_buffer_row`, rather than a secret flag shoved onto the end of the character data, since it's not actually character data at all.

Reply:
Yes it's a good idea, I changed it.